jetson/face_extractor.py

```diff
@@ -1,56 +1,35 @@
 import argparse
 from glob import glob
-import hashlib
 import os
 import warnings
 
 import cv2
 import numpy as np
 import torch
-from PIL import Image
 from tqdm import tqdm
 
-from face_detector import FaceDetector
+from face_detector_threaded import FaceDetector
 
 warnings.filterwarnings('once')
+IMAGE_EXT = ['.jpg', '.JPG', '.png', '.PNG']
+VIDEO_EXT = ['.mp4', '.MP4', 'mov', '.MOV']
 
 
-# for videos, saves filenames and hashes
-# for images, saves filenames
-def get_files(input_dir, hash_file=None, images=False):
-    if images:
-        img_ext = ['.jpg', '.JPG', '.png', '.PNG']
-        imgs = [glob(f"{input_dir}/*{e}", recursive=args.recursive) for e in img_ext]
-        images = [img for subimg in imgs for img in subimg]
-        return images
-    else:
-        video_ext = ['.mp4', '.mov', '.MOV', '.MP4']
-        videos = [glob(f"{input_dir}/*{e}", recursive=args.recursive) for e in video_ext]
-        # Reduce the 2d list from before to a 1d list
-        videos = [vid for subvid in videos for vid in subvid]
-
-        hashes = []
-        for video in videos:
-            hasher = hashlib.md5()
-            buf_size = 65536
-            with open(video, 'rb') as video_file:
-                buf = video_file.read(buf_size)
-                while len(buf) > 0:
-                    hasher.update(buf)
-                    buf = video_file.read(buf_size)
-            hashes.append(hasher.hexdigest())
-
-        if hash_file is None:
-            return list(zip(videos, hashes))
-        else:
-            with open(hash_file, 'r') as f:
-                processed_hashes = [line.strip() for line in f]
-            # Filter out the videos that have already been processed
-            return [(file, file_hash) for (file, file_hash) in zip(videos, hashes) if file_hash not in processed_hashes]
+def get_images():
+    """Get filenames of images from the input directory."""
+    files = [glob(f"{args.input_dir}/*{e}") for e in IMAGE_EXT]
+    return files[0]
+
+
+def get_videos():
+    """Get filenames of videos from the input directory."""
+    files = [glob(f"{args.input_dir}/*{e}") for e in VIDEO_EXT]
```

Comment:
yeah make `input_dir` a parameter please. This way might work but doesn't scale well and can lead to issues in the future